js/query.js

```diff
@@ -1,70 +1,205 @@
-                    function(response) {
-                        if (response.status !== 200) {
-                            console.log('Looks like there was a problem. Status Code: ' +
-                                response.status);
-                            return;
-                        }
+    function hideData(divObj) {
+        // Hide all data in child nodes of givin div element.
+        var labels = divObj.querySelectorAll("label");
+        var links = divObj.querySelectorAll('a');
+        var i;
+        var textDivs = divObj.getElementsByClassName("text");
+
+        for (i = 0; i < labels.length; i++) {
+            var childData = labels[i];
+
+            if (!(childData.classList.contains('hidden'))) {
+                childData.classList.add('hidden');
+            }
+        }
+
+        for (i = 0; i < textDivs.length; i++) {
+            var childData = textDivs[i];
+
+            if (!(childData.classList.contains('hidden'))) {
+                childData.classList.add('hidden');
+                childData.textContent = '';
+            }
+        }
+
+        for (i = 0; i < links.length; i++) {
+            var childData = links[i];
+            childData.remove();
+        }
+    }
 
-                        // Examine the text in the response
-                        response.json().then(function(data) {
-                            console.log(data);
-                            var summary = data.summary
-                            document.getElementById('summary').textContent = 'Gene Summary: ' + summary;
-                        });
+    function annotateGene(gene) {
+        fetch('https://mygene.info/v3/gene/' + gene)
+            .then(
+                function(response) {
+                    if (response.status !== 200) {
+                        console.log('Looks like there was a problem. Status Code: ' +
+                            response.status);
+                        return;
                     }
-                )
-                .catch(function(err) {
-                    console.error('Fetch Gene Annotation Error', err);
-                });
+
+                    // Examine the text in the response
+                    response.json().then(function(data) {
+                        console.log(data);
+                        var info = parseGeneData(data);
+                        displayData(info);
+                    });
+                }
+            )
+            .catch(function(err) {
+                console.error('Fetch Gene Annotation Error', err);
+            });
+    }
+
+    function parseGeneData(data) {
+        var aliases;
+        var hgnc;
+        var coords;
+
+        if (data.hasOwnProperty('HGNC')) {
+            hgnc = { 'db': 'https://www.genenames.org/cgi-bin/gene_symbol_report?hgnc_id=', 'ident': data.HGNC };
+        } else {
+            hgnc = undefined;
+        }
+
+        if (data.hasOwnProperty('alias') && typeof data.alias == 'string') {
```

Comment:
Congratulations, for you have stumbled upon one of JavaScript's many bugbears.

[Read this.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comparison_Operators#Equality_operators)

I suggest always using strict equality or inequality. So, replace `==` with `===`.